core/src/main/java/org/apache/iceberg/BaseTransaction.java

```diff
@@ -241,7 +234,14 @@ private void commitCreateTransaction() {
     }
   }
 
-  private void commitReplaceTransaction() {
+  private void commitReplaceTransaction(boolean orCreate) {
+    if (base == null && !orCreate) {
+      throw new NoSuchTableException("Table doesn't exist");
+    } else if (base == null) {
+      commitCreateTransaction();
```

Comment:
I don't think this is correct because `commitCreateTransaction` won't succeed if the table is created concurrently, but create or replace should succeed.

Reply:
Should we use `replaceMetadata` to get the number of retries if `base` is null?

Reply:
If `base` is null, I think we can either use `replaceMetadata` instead of `base` and keep the existing try-catch or we can actually call `commitCreateTransaction` wrapped in another try first. If we get `AlreadyExistsException`, we can call the second one, which will replace the table.

Reply:
Isn't base going to be reinitialized correctly just before the commit? I think the right place to do this is in retry.

Reply:
We need to know how many retries to use, etc. We cannot configure `Tasks` as `base` is null.
```
Tasks.foreach(ops)
     .retry(base.propertyAsInt(COMMIT_NUM_RETRIES, COMMIT_NUM_RETRIES_DEFAULT))
     .exponentialBackoff(
         base.propertyAsInt(COMMIT_MIN_RETRY_WAIT_MS, COMMIT_MIN_RETRY_WAIT_MS_DEFAULT),
         base.propertyAsInt(COMMIT_MAX_RETRY_WAIT_MS, COMMIT_MAX_RETRY_WAIT_MS_DEFAULT),
         base.propertyAsInt(COMMIT_TOTAL_RETRY_TIME_MS, COMMIT_TOTAL_RETRY_TIME_MS_DEFAULT),
         2.0 /* exponential */)
     .onlyRetryOn(CommitFailedException.class)
     .run(underlyingOps -> {
       ...
     }
```

Reply:
In that case, I think we should use the settings from the current metadata instead of base.

Reply:
Done